online-fishing-store/src/app/pages/profile-page/favorite-products/favorite-products.component.html

```diff
@@ -0,0 +1,9 @@
+<div class="favorite-products" *ngFor="let favoriteProduct of favoriteProducts">
+    <div>
+        <app-favorite-products-item
+            [favoriteProduct]="favoriteProduct"
+            (click)="removeFromFavorites(favoriteProduct)" 
```

Comment:
You have selectedFavoriteProduct event in this item. You should use this event instead of "click" event